README.md

```diff
@@ -1,3 +1,79 @@
+
+The Kinobi IDL is designed as a tree of nodes starting with the `RootNode` which contains a `ProgramNode` and additional data such as the Kinobi version used when the IDL was created. Kinobi provides over 60 different types of nodes that help describe just about any aspect of your Solana programs. [You can read more about the Kinobi nodes here](./packages/nodes).
+
+![A small example of a Kinobi IDL as a tree of nodes. It starts with a RootNode and goes down to ProgramNode, AccountNode, InstructionNode, etc.](https://github.com/kinobi-so/kinobi/assets/3642397/fc89615f-f881-4160-a0a1-31b6d3da26da)
+
+Because everything is designed as a `Node`, we can transform the IDL, aggregate information and output various utility tools using special objects that can traverse node trees known as visitors. [See this documentation to learn more about Kinobi visitors](./packages/visitors).
+
+![A small example of how a visitor can transform a Kinobi IDL into another Kinobi IDL. This example illustrates the "deleteNodesVisitor" which recursively removes NumberTypeNodes from a tree of nested TypleTypeNodes.](https://github.com/kinobi-so/kinobi/assets/3642397/958efd93-e866-4e6d-a99f-fb18606b58c1)
+
+## From program to Kinobi
+
+There are various ways to extract information from your Solana programs in order to obtain a Kinobi IDL.
+
+-   **Using Kinobi macros**. This is not yet available but you will soon have access to a set of Rust macros that help attach IDL information directly within your Rust code. These macros enable Kinobi IDLs to be generated whenever you build your programs.
+-   **From Anchor IDLs**. If you are using [Anchor programs](https://github.com/coral-xyz/anchor) or [Shank macros](https://github.com/metaplex-foundation/shank), then you can get an Anchor IDL from them. You can then use the `@kinobi-so/nodes-from-anchor` package to convert that IDL into a Kinobi IDL as shown in the code snippet below. Note that the Anchor IDL might not offer all the information that Kinobi can hold and therefore, you may want to transform your Kinobi IDL to provide additional information. You can learn more about this in the next section.
+
+    ```ts
+    import { createFromRootNode } from 'kinobi';
+    import { rootNodeFromAnchor } from '@kinobi-so/nodes-from-anchor';
+    import anchorIdl from 'anchor-idl.json';
+
+    const kinobi = createFromRootNode(rootNodeFromAnchor(anchorIdl));
+    ```
+
+-   **By hand**. If your Solana program cannot be updated to use Kinobi macros and you don’t have an Anchor IDL, you may design your Kinobi IDL by hand. We may provide tools such as a Kinobi Playground to help with that in the future.
+
+## Transforming Kinobi
+
+Once you have your Kinobi IDL, you may use visitors to transform it further. This can be useful when the Kinobi IDL was obtained from another source that may not contain as much information. Here is an example using two provided visitors that adjust the accounts and instructions on the program.
```

Comment:
```suggestion
Once you have your Kinobi IDL, you may use visitors to transform it further. This can be useful when the Kinobi IDL was obtained from another source that may not contain as much information. Here is an example using two provided visitors that adjusts the accounts and instructions on the program.
```